### PR TITLE
Add a concept page for each API resource

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Add a concept page for each API resource and a glossary for commonly used terms.

#### Which issue(s) this PR fixes:

Part of #64

### CODE REVIEW

**Comment by alculquicondor** [comment]
/hold
to prevent accidental premature merge

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

File: docs/concepts/cluster_queue.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Cluster Queue
+
+A `ClusterQueue` is a cluster-scoped resource that governs a pool of resources
+such as CPU, memory and hardware accelerators. A `ClusterQueue` defines:
+- The resource _flavors_ that it manages, with usage limits and order of consumption.
+- Fair sharing rules across the tenants of the cluster.
+
+Only cluster administrators should create `ClusterQueue` objects.
```
Comment (ahg-g) [inline docs/concepts/cluster_queue.md:8]: lets also add something about personas (Batch Admin and Batch User), perhaps in glossary too

File: docs/concepts/cluster_queue.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## Resource Flavors
+
+Resources in a cluster are typically not homogeneus. Resources could differ in:
+- pricing and availability (ex: spot vs on-demand VMs)
+- architecture (ex: x86 vs ARM CPUs)
+- brands and models (ex: Radeon 7000 vs Nvidia A100 vs T4 GPUs)
+
+A `ResourceFlavor` is a resource that represents these variations and allows
```
Comment (ahg-g) [inline docs/concepts/cluster_queue.md:17]:
```suggestion
A `ResourceFlavor` is an object API that represents these variations and allows
```

just so it is not confused with compute resource

File: docs/concepts/cluster_queue.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+administrators to associate them with node labels and taints.
+
+## Cohort
+
+`ClusterQueues` can be grouped in _cohorts_. `ClusterQueues` that belong to the
+same cohort can borrow unused resources from each other, if they have matching
```
Comment (ahg-g) [inline docs/concepts/cluster_queue.md:23]:
```suggestion
same cohort can borrow unused quota from each other, if they have matching
```

File: README.md [hunk 4]
```diff
@@ -1,14 +1,23 @@
...
+and well-established third-party controllers. Autoscaling, pod-to-node scheduling and
+job lifecycle management are the responsibility of cluster-autoscaler,
+kube-scheduler and kube-controller-manager, respectively. Advanced
+admission control can be delegated to controllers such as [gatekeeper](https://github.com/open-policy-agent/gatekeeper).
+
+<!-- TODO(#64) Remove links to google docs once the contents have been migrated to this repo -->
+[bit.ly/kueue-apis](https://bit.ly/kueue-apis) (please join the [mailing list](https://groups.google.com/a/kubernetes.io/g/wg-batch)
+to get access) discusses the API proposal and a high-level description of how it
+operates; while [bit.ly/kueue-controller-design](https://bit.ly/kueue-controller-design)
+presents the detailed design of the controller.
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply README.md]: Done

File: docs/concepts/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+The process of admitting a workload to start (pods to be created). A workload
+is admitted by a ClusterQueue according to the available resources and gets
+resource flavors assigned for each requested resource. Sometimes referred to
+as _workload scheduling_ or _job scheduling_ (not to be confused with
+[pod scheduling](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/)).
+
+### [Cohort](cluster_queue.md#cohort)
+
+A group of cluster queues that can borrow unused resources from each other.
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply docs/concepts/README.md]: Done

File: docs/concepts/README.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+### Queueing
+
+The time between a workload is created until it is admitted by Cluster Queue.
+Typically, the workload will compete with other workloads for available
+resources based on the fair sharing rules for the Cluster Queue.
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply docs/concepts/README.md]: Done

File: docs/concepts/README.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+as _workload scheduling_ or _job scheduling_ (not to be confused with
+[pod scheduling](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/)).
+
+### [Cohort](cluster_queue.md#cohort)
+
+A group of cluster queues that can borrow unused resources from each other.
+
+### Queueing
+
+The time between a workload is created until it is admitted by Cluster Queue.
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply docs/concepts/README.md]: Done

File: docs/concepts/cluster_queue.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
+# Cluster Queue
+
+A `ClusterQueue` is a cluster-scoped resource that governs a pool of resources
+such as CPU, memory and hardware accelerators. A `ClusterQueue` defines:
+- The resource _flavors_ that it manages, with usage limits and order of consumption.
+- Fair sharing rules across the tenants of the cluster.
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply docs/concepts/cluster_queue.md:6]: I'll leave it for a follow up. I need more details in the ClusterQueue page first.

File: docs/concepts/cluster_queue.md (hunk 1 above)
Comment (alculquicondor) [reply docs/concepts/cluster_queue.md:8]: I was thinking about that too. I'll do it in a follow up.

File: docs/concepts/cluster_queue.md (hunk 2 above)
Comment (alculquicondor) [reply docs/concepts/cluster_queue.md]: Done, updated everywhere.

File: docs/concepts/cluster_queue.md (hunk 3 above)
Comment (alculquicondor) [reply docs/concepts/cluster_queue.md]: Done

**Pull request merged**